### PR TITLE
FIX ISSUE-140 by using response for returning value

### PR DESCRIPTION
**Pull request by SuperheroicCoding**
@see https://github.com/salemdar/ngx-cookie/issues/140

Using response header set cookie to return cookie value on Server

### CODE REVIEW

File: projects/ngx-cookie-backend/src/lib/cookie-backend-writer.service.ts [hunk 1]
```diff
@@ -1,23 +1,56 @@
...
-
-  constructor(@Optional() @Inject(REQUEST) private request: Request,
-              @Optional() @Inject(RESPONSE) private response: Response) {}
+  constructor(
+    @Optional() @Inject(REQUEST) private request: Request,
+    @Optional() @Inject(RESPONSE) private response: Response,
+  ) {
+  }
 
   readAllAsString(): string {
```
Comment (salemdar) [inline projects/ngx-cookie-backend/src/lib/cookie-backend-writer.service.ts:16]: Could you remove blank lines in methods?

File: projects/ngx-cookie-backend/src/lib/cookie-backend-writer.service.ts [hunk 2]
```diff
@@ -1,23 +1,56 @@
...
+    const cookiesFromRequest: string[] = requestHeadersCookies ? requestHeadersCookies.split(COOKIE_SEPARATOR) : [];
+
+    const addedCookies: string[] = this.getNormalizedResponseCookies();
+
+    const allCookies = this.latestUniqueCookieValues(cookiesFromRequest, addedCookies);
+    return allCookies.join(COOKIE_SEPARATOR);
+  }
+
+  private getNormalizedResponseCookies(): string[] {
+    const responseCookies: string[] | string = (this.response.getHeader('Set-Cookie') as any) ?? '';
```
Comment (salemdar) [inline projects/ngx-cookie-backend/src/lib/cookie-backend-writer.service.ts:28]: Can't we avoid `any` here?

File: projects/ngx-cookie-backend/src/lib/cookie-backend-writer.service.ts [hunk 3]
```diff
@@ -33,15 +66,14 @@ export class CookieBackendWriterService implements ICookieWriterService {
       path: options?.path,
       domain: options?.domain,
       secure: options?.secure,
-      sameSite: options?.sameSite
+      sameSite: options?.sameSite,
```
Comment (salemdar) [inline projects/ngx-cookie-backend/src/lib/cookie-backend-writer.service.ts:69]: Unnecessary comma.

File: projects/ngx-cookie-backend/src/lib/cookie-backend-writer.service.ts [hunk 4]
```diff
@@ -1,23 +1,56 @@
...
+import { CookieOptions, ICookieWriterService, isEmpty, isNil, isString } from 'ngx-cookie';
 
+const COOKIE_SEPARATOR = '; ';
 
 @Injectable()
 export class CookieBackendWriterService implements ICookieWriterService {
-
-  constructor(@Optional() @Inject(REQUEST) private request: Request,
-              @Optional() @Inject(RESPONSE) private response: Response) {}
+  constructor(
```
Comment (salemdar) [inline projects/ngx-cookie-backend/src/lib/cookie-backend-writer.service.ts:10]: Let's stick to previous indentation.

File: projects/ngx-cookie-backend/src/lib/cookie-backend-writer.service.ts [hunk 5]
```diff
@@ -1,23 +1,56 @@
...
+
+    const allCookies = this.latestUniqueCookieValues(cookiesFromRequest, addedCookies);
+    return allCookies.join(COOKIE_SEPARATOR);
+  }
+
+  private getNormalizedResponseCookies(): string[] {
+    const responseCookies: string[] | string = (this.response.getHeader('Set-Cookie') as any) ?? '';
+
+    const addedCookies: string[] =
+      responseCookies instanceof Array
```
Comment (salemdar) [inline projects/ngx-cookie-backend/src/lib/cookie-backend-writer.service.ts:31]: I like utility methods where possible: `Array.isArray(responseCookies)`

File: projects/backend-test-app/src/app/app.component.ts [hunk 6]
```diff
@@ -27,7 +28,10 @@ export class AppComponent implements OnInit {
   }
 
   ngOnInit(): void {
-    this.setCookies();
+    if (isPlatformServer(this.platformId)) {
```
Comment (salemdar) [inline projects/backend-test-app/src/app/app.component.ts:31]: Good idea 👍

**Comment by SuperheroicCoding** [comment]
@salemdar What's the current state? I don't know how to fix the e2e tests. It's a CI problem I think.

**Review by salemdar — CHANGES_REQUESTED** [review]
Thanks for the good work. I made some comments. It would be very cool if you can check them.

You also need to merge with master to make e2e tests running.

**Comment by salemdar** [comment]
Hey @SuperheroicCoding ,
Did you have any chance to check my comments?

**Comment by SuperheroicCoding** [comment]
Hi, just saw your comments. Feel free to adapt it yourself. Because I am currently quite busy. Earliest next week I can work again on it.

File: projects/ngx-cookie-backend/src/lib/cookie-backend-writer.service.ts (hunk 3 above)
Comment (SuperheroicCoding) [reply projects/ngx-cookie-backend/src/lib/cookie-backend-writer.service.ts]: removed

File: projects/ngx-cookie-backend/src/lib/cookie-backend-writer.service.ts (hunk 4 above)
Comment (SuperheroicCoding) [reply projects/ngx-cookie-backend/src/lib/cookie-backend-writer.service.ts]: fixed

**Comment by SuperheroicCoding** [comment]
@salemdar Finally found time to include your changes. Please have a look again.

**Comment by SuperheroicCoding** [comment]
@salemdar Can  you please check again? 
Can you give me hint for fixing the failing e2e tests?

**Review by salemdar — APPROVED** [review]

**Pull request merged**